keps/0027-force-pod-reload-parameters.md

```diff
@@ -0,0 +1,203 @@
+---
+kep-number: 27
+short-desc: Detailed control what parameter changes trigger pod restarts
+title: Pod restart control
+authors:
+  - @aneumann82
+owners:
+  - @aneumann82
+editor:
+creation-date: 2020-03-31
+last-updated: 2020-04-15
+status: provisional
+see-also:
+replaces:
+superseded-by:
+---
+
+## Summary
+
+This KEP describes a feature to allow detailed control when templated pods in a stateful set are restarted when parameters
+that are not used in the template change.
+
+## Motivation
+
+KUDO currently has a feature that automatically restarts Pods of Deployments or StatefulSets if any KUDO-Parameter is changed. This
+is often required as a Parameter can change a value in a ConfigMap that is used by the Pods. If the Pods are not restarted
+the change is propagated to the Pods, but as most applications are not aware of changing config files they require a 
+restart. This was originally solved by using [stakater/Reloader](https://github.com/stakater/Reloader) but this had 
+[issues](https://github.com/kudobuilder/kudo/issues/1025) which lead to the self developed solution, which is at the 
+moment applied to [all templates](https://github.com/kudobuilder/kudo/issues/1036)
+
+This behavior can currently not be controlled and is always active. The issue is that not all Parameter changes require
+a rolling restart of Pods, and sometimes the restart is unwanted and can negatively affect the state of the application
+the operator is controlling.
+
+One example would be the `replica` count of a StatefulSet: An increase here should only start new Pods and not restart
```

Comment:
do we have more examples other than `replica`? I feel we just have this one specific use case of scaling up and down sts/deployment.

in the case of another example, where an application regularly re-reads the config files, the `trigger` plan of the CM shouldn't include in any of its tasks the sts/deployment. That won't trigger any update.